sdks/java/core/src/main/java/org/apache/beam/sdk/values/DrainMode.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.values;
+
+/** {@link DrainMode} indicates whether a pipeline is known to be draining or not. */
+public enum DrainMode {
+  UNSPECIFIED,
```

Comment:
Now that I have gotten this far into implementation, I can see that the SDK does not need this value. It is necessary in protobuf but should be converted to `NOT_DRAINING` because we want to never encode it unless it is draining.